google-cloud-storage/src/main/java/com/google/cloud/storage/Option.java

```diff
@@ -16,53 +16,17 @@
 
 package com.google.cloud.storage;
 
-import static com.google.common.base.Preconditions.checkNotNull;
-
-import com.google.cloud.storage.spi.v1.StorageRpc;
-import com.google.common.base.MoreObjects;
+import com.google.cloud.storage.UnifiedOpts.Opt;
 import java.io.Serializable;
-import java.util.Objects;
 
 /** Base class for Storage operation option. */
-public abstract class Option implements Serializable {
-
-  private static final long serialVersionUID = -73199088766477208L;
-
-  private final StorageRpc.Option rpcOption;
-  private final Object value;
-
-  Option(StorageRpc.Option rpcOption, Object value) {
-    this.rpcOption = checkNotNull(rpcOption);
-    this.value = value;
-  }
-
-  StorageRpc.Option getRpcOption() {
-    return rpcOption;
-  }
-
-  Object getValue() {
-    return value;
-  }
+@Deprecated
+public abstract class Option<O extends Opt> extends UnifiedOpts.OptionShim<O>
```

Comment:
I was wondering how Option<T> was actually going to work; lucky for us it's a class in our library.

Reply:
Yeah, thankfully it's not in core